roles/matrix-ntfy/defaults/main.yml

```diff
@@ -14,6 +14,13 @@ matrix_ntfy_docker_image_force_pull: "{{ matrix_ntfy_docker_image.endswith(':lat
 # Public facing base URL of the ntfy service
 matrix_ntfy_base_url: "https://{{ matrix_server_fqn_ntfy }}"
 
+# Rate limits
+matrix_ntfy_limit_burst_rate: 60 # default
+matrix_ntfy_limit_replenish_rate: "5s" # default
+matrix_ntfy_limit_rate_exempt: "{{matrix_server_fqn_matrix}},localhost" # exempt our matrix server from rate limits, this may not work when the homeserver's outgoing IP is different from the incoming IP, but most small deployments should be fine.
```

Comment:
How does the homeserver currently hit Ntfy? Via `matrix_ntfy_base_url`? Via container URLs like `http://matrix-ntfy:PORT`?

As per our `examples/hosts`, we ask people to use an external IP address for `ansible_host`: https://github.com/spantaleev/matrix-docker-ansible-deploy/blob/d159408feb0888dce1110d65955e1e4916c44739/examples/hosts#L1-L2

We pass this to `matrix_coturn_turn_external_ip_address` here: https://github.com/spantaleev/matrix-docker-ansible-deploy/blob/d159408feb0888dce1110d65955e1e4916c44739/group_vars/matrix_servers#L1413

You can do something similar:

- introduce a new `matrix_ntfy_external_ip_address` variable in this `defaults/main.yml` file, set to empty
- add a `matrix_ntfy_external_ip_address: "{{ ansible_host }}"` definition to the `matrix-ntfy` section in `group_vars/matrix_servers`
- use `matrix_ntfy_external_ip_address` (if not empty) in `matrix_ntfy_limit_rate_exempt`

Not sure if this is necessary or helpful though. `matrix_ntfy_limit_rate_exempt` seems to use hostnames, not IP addresses. It likely resolves these domain names correctly, so `matrix_server_fqn_matrix` would end up reliably pointing to the external IP address of the deployment. There's probably no need to improve on that.

Reply:
> How does the homeserver currently hit Ntfy? Via `matrix_ntfy_base_url`? Via container URLs like `http://matrix-ntfy:PORT`?

The homeserver reaches ntfy the same way that a phone reaches ntfy, via the public host ntfy.BASE_DOMAIN. It can't tell the difference between a ntfy instance hosted by this playbook or one on another server.

Reply:
> Not sure if this is necessary or helpful though. `matrix_ntfy_limit_rate_exempt` seems to use hostnames, not IP addresses. It likely resolves these domain names correctly, so `matrix_server_fqn_matrix` would end up reliably pointing to the external IP address of the deployment. There's probably no need to improve on that.

It can use both hostnames and IP addresses. I am setting up a temporary server for testing rn, and will see if it properly detects it.

Reply:
Also, clarification: I added localhost here just in case some weird thing happens with IP detection since both services are running on the same machine. It is unlikely, since nginx would get in the way, but there's no harm in adding it.

Reply:
Sure! No harm in keeping it.